src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -21,12 +23,14 @@
     private final Logger logger = LogsCenter.getLogger(LogicManager.class);
 
     private final Model model;
+    private final Storage storage;
     private final CommandHistory history;
     private final AddressBookParser addressBookParser;
     private final UndoRedoStack undoRedoStack;
 
-    public LogicManager(Model model) {
+    public LogicManager(Model model, Storage storage) {
         this.model = model;
+        this.storage = storage;
```

Comment:
Can we do a _debouncing_ here? Meaning that, Logic component should not deal with storage directly. Instead, it should ask model component to handle the transaction.

Reply:
But then model will interface with storage, where it previously doesn't need to. Saving after executing Undoable commands are done via the Events Centre through model, where I don't think it is appropriate because in export, it is not a side effect. Hence, I think the export command should directly interface Storage.

Reply:
Yes, so for those undoable commands, it is still `Model` rather than `Logic` that is dealing with `Storage` (indirectly though, through `EventCenter`). But nvm, this is low priority.